osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs

```diff
@@ -63,13 +63,14 @@ protected override void CreateContent()
             for (int i = 0; i < requiredCircles; i++)
             {
                 float diameter = (offset + (i + 1) * DistanceBetweenTicks) * 2;
+                const float thickness = 4;
 
                 AddInternal(new Ring(ReferenceObject, GetColourForIndexFromPlacement(i))
                 {
                     Position = StartPosition,
                     Origin = Anchor.Centre,
-                    Size = new Vector2(diameter),
-                    InnerRadius = 4 * 1f / diameter,
+                    Size = new Vector2(diameter + thickness / 2),
+                    InnerRadius = thickness * 1f / diameter,
```

Comment:
So I went to fix the test failures, expecting that it's just a matter of a test being overly clingy wrt implementation, but upon inspection this seems to be actually wrong?

Explanation: I believe that I understand the intent of adding `thickness / 2`is that it is done in order to enlarge the `Ring` by `thickness / 2` on both sides, therefore ensuring that the center of the `Ring` lands precisely on the exact diameter of the snap. However, note that `Size` is effectively the _diameter_ and not the _radius_. So you would have to add `thickness / 2 * 2` to achieve this. Additionally, `InnerRadius` is not fully correct as it is a value relative to the _post-adjusted_ diameter of the ring.

All in all, I believe that a more correct version of this looks something like:

```diff
diff --git a/osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs b/osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs
index f27ecf60cc..b59d4638cd 100644
--- a/osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs
+++ b/osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs
@@ -62,14 +62,14 @@ protected override void CreateContent()
 
             for (int i = 0; i < requiredCircles; i++)
             {
-                float diameter = (offset + (i + 1) * DistanceBetweenTicks) * 2;
                 const float thickness = 4;
+                float diameter = (offset + (i + 1) * DistanceBetweenTicks + thickness / 2) * 2;
 
                 AddInternal(new Ring(ReferenceObject, GetColourForIndexFromPlacement(i))
                 {
                     Position = StartPosition,
                     Origin = Anchor.Centre,
-                    Size = new Vector2(diameter + thickness / 2),
+                    Size = new Vector2(diameter),
                     InnerRadius = thickness * 1f / diameter,
                 });
             }
```

Reply:
I did understand that `InnerRadius` is relative, and originally had something closer to your proposal but couldn't figure why it wasn't working (it was making the line unexpectedly thicker and also not centring correctly). I'll check your proposal and see if it looks better.

Reply:
Change looks solid, have applied 👍 .